### DISCUSSION TITLE
What is the principle of vue declaration file generation

### DISCUSSION THREAD

**Discussion by TanxiangCode** [element-plus/element-plus#1362] [Q&A]
In the lib folder, I found that there are many declaration files similar to XXX.vue.d.ts. I want to ask how this is generated from the vue file.

**Comment by jw-foss** [comment]
It was generated by `rollup`.

**Reply by TanxiangCode** [reply]
我运行了命令“build:esm-bundle”，并且运行成功，但并没有生成vue.d.ts
![image](https://user-images.githubusercontent.com/42427662/105833727-93c18100-6004-11eb-9f8c-700beef0669c.png)
例如，没有"./src/button.vue.d.ts"这个文件
![image](https://user-images.githubusercontent.com/42427662/105833785-a50a8d80-6004-11eb-92c0-21fa1b352c90.png)

**Reply by jw-foss** [reply]
If you'd like to know the detail, please go to `rollup-plugin-typescript` to check it out. I can offer no further help.

**Comment by jw-foss** [comment]
Closing since this is irrelevant

**Comment by jw-foss** [comment]
BTW, discussion is meant to discuss things like `feature request` `API design` instead of the project detail like this. If you are interested in the project detail, please go see the source code. It's an OSS project.
